hash.go

```diff
@@ -40,8 +40,18 @@ func (h Hash) String() string {
 }
 
 // Sum builds a digest according to the specs in the Holochain
-func (h *Hash) Sum(hc HashSpec, data []byte) (err error) {
-	h.H, err = mh.Sum(data, hc.Code, hc.Length)
+func (hash *Hash) Sum(holo *Holochain, data []byte) (err error) {
+	coding := holo.WireType
+	Debugf("Sum %v <%v> %v\n", coding, coding == WIRE_GOB, WIRE_GOB)
+	switch coding {
+	case WIRE_GOB:
+		hspec := holo.HashSpec
+		hash.H, err = mh.Sum(data, hspec.Code, hspec.Length)
+	case WIRE_JSON:
+		err = errors.New("WIRE_JSON not implemented")
+	default:
+		err = errors.New("Bad coding " + coding)
+	}
```

Comment:
The coding switch logic isn't implemented everywhere. If you want I think go's somewhat limited inheritance mechanisms to use Holochain subclassing instead having to test attribute and dispatch in all the methods. Using interfaces could be another good option too. Something like how conversion of general objects to strings with an interface would be good.